src/sync/sync-utils.ts

```diff
@@ -52,3 +59,42 @@ export const getCommitSinceDate = async (jiraHost: string, flagName: NumberFlags
 	}
 	return new Date(Date.now() - timeCutoffMsecs);
 };
+
+const getGitHubAppConfig = async (subscription: Subscription, logger: Logger): Promise<GitHubAppConfig> => {
+
+	const gitHubAppId = subscription.gitHubAppId;
+
+	//cloud
+	if (!gitHubAppId) return cloudGitHubAppConfig();
+
+	//ghes
+	const gitHubServerApp = await GitHubServerApp.findByPk(gitHubAppId);
+	if (!gitHubServerApp) {
+		logger.error("Cannot find gitHubServerApp by pk", { gitHubAppId: gitHubAppId, subscriptionId: subscription.id });
+		throw new Error("Error duing find and start sync. Reason: Cannot find ghes record from subscription.");
+	}
+	return ghesGitHubAppConfig(gitHubServerApp);
+
+};
+
+const cloudGitHubAppConfig = () => {
+	return {
+		gitHubAppId: undefined,
+		appId: parseInt(envVars.APP_ID),
+		clientId: envVars.GITHUB_CLIENT_ID,
+		gitHubBaseUrl: GITHUB_CLOUD_HOSTNAME,
+		gitHubApiUrl: GITHUB_CLOUD_API_BASEURL,
+		uuid: undefined
+	};
+};
+
+const ghesGitHubAppConfig = (app: GitHubServerApp): GitHubAppConfig => {
+	return {
+		gitHubAppId: app.id,
+		appId: app.appId,
+		uuid: app.uuid,
+		clientId: app.gitHubClientId,
+		gitHubBaseUrl: app.gitHubBaseUrl,
+		gitHubApiUrl: app.gitHubBaseUrl
+	};
+};
```

Comment:
hm, I have a feeling that this config creation should be abstracted somewhere so anything can use it.

Reply:
I was thinking exactly the same thing. But didn't do it in this PR, as currently there're three major places to have this config:

1. WebhookReceiverPost  --> no secrets, just id --> config pushed to context --> queue payload
2. GitHubServerMiddlware, --> secrets also available --> config saved in res.locals
3. uitl/get-github-client-config --> private keys available --> config used locally to createXXXClient.

And shapes are also a bit different as well.

Maybe let's discuss it and address all above in another PR?

Reply:
sounds good :+1:

Reply:
Ticket created. 
https://softwareteams.atlassian.net/browse/ARC-1582